includes/admin/class.llms.admin.settings.php

```diff
@@ -982,74 +976,125 @@ public static function save_fields( $settings ) {
 				case 'radio':
 				case 'hidden':
 				case 'image':
-					if ( isset( $_POST[ $value['id'] ] ) ) {
-						$option_value = llms_filter_input_sanitize_string( INPUT_POST, $value['id'] );
+					if ( $is_array_option ) {
+						$option_value = self::get_array_field_posted_value( $field['id'] );
+					} elseif ( isset( $_POST[ $field['id'] ] ) ) {
+						$option_value = llms_filter_input_sanitize_string( INPUT_POST, $field['id'] );
 					} else {
 						$option_value = '';
 					}
 
-					if ( isset( $value['sanitize'] ) && 'slug' === $value['sanitize'] ) {
+					if ( isset( $field['sanitize'] ) && 'slug' === $field['sanitize'] ) {
 						$option_value = sanitize_title( $option_value );
 					}
 
 					break;
 
 				case 'multiselect':
-					if ( isset( $_POST[ $value['id'] ] ) ) {
-						$option_value = llms_filter_input_sanitize_string( INPUT_POST, $value['id'], array( FILTER_REQUIRE_ARRAY ) );
+					if ( isset( $_POST[ $field['id'] ] ) ) {
+						$option_value = llms_filter_input_sanitize_string( INPUT_POST, $field['id'], array( FILTER_REQUIRE_ARRAY ) );
 					} else {
 						$option_value = '';
 					}
 					break;
 
-				// Custom handling.
 				default:
-					do_action( 'lifterlms_update_option_' . $type, $value );
-
-					break;
+					/**
+					 * Action run for external field types.
+					 *
+					 * @since Unknown
+					 * @deprecated [version] Use `llms_update_option_{$type}` filter hook instead.
+					 *
+					 * @param type $arg Description.
+					 */
+					do_action_deprecated( "lifterlms_update_option_{$type}", array( $field ), '[version]' );
```

Comment:
I guess you haven't added the "replacement" here because it's not a true replacement since the new hook is a filter hook. Right?

Reply:
yea i thought three things when I read through here:

Oh m**k...
That should be a filter
It should run against all types, not just custom types.

Realistically the action works just fine and doesn't need to be deprecated. But it will require a developer to perform their own save logic and call update_action() on their own in favor of just returning a value and letting us do it.

Reply:
Yeah.

Reply:
Sorry, I meant just:
Yeah